File: js/tlg/state/StateHighscore.ts
import { State } from './State';
import { InternalContext } from '../context/InternalContext';
import { PlatformContext } from '../context/PlatformContext';
import { HighscoreList } from '../score/HighscoreList';
import { StateID } from './StateID';
import { StorageContext } from '../context/StorageContext';

export class StateHighscore extends State {
    
    constructor(c:InternalContext, s:StorageContext) {
        super(c,s);
        this.context.setStatusText("New Highscore!");
    }

    
    public init(c:PlatformContext):StateID {
        // c.onNewHighscore();
        return super.init(c);
    }

    
    
    public setHighscoreName(c:PlatformContext, name:string):StateID {

        if (name.length>0) {
            let highscoreList = new HighscoreList(c);

            highscoreList.add(name, this.context.currentScore.getScore());
            highscoreList.writeState(c);

            return new StateID(StateID.ID_LOCKED);

        } else {
            return super.setHighscoreName(c,name);
        }
    }

    
    public getID() {
        return StateID.ID_HIGHSCORE;
    }
}

File: tlg/src/main/kotlin/state/StateHighscore.kt
package state


import context.InternalContext
import context.PlatformContext
import score.HighscoreList
import java.io.IOException

class StateHighscore(c: InternalContext) : State(c) {
    override fun init(c: PlatformContext): State {
        context.state = this
        c.onNewHighscore()
        return this
    }

    override fun setHighscoreName(c: PlatformContext, name: String): State {
        return if (name.isNotEmpty()) {
            val highscoreList = HighscoreList(c)
            highscoreList.add(name, context.currentScore.score)
            try {
                highscoreList.writeState(c)
            } catch (e: IOException) {
                e.printStackTrace()
            }
            StateLocked(context).init(c)
        } else {
            this
        }
    }

    override fun toString(): String {
        return "New Highscore!"
    }

    override val id = ID

    companion object {
        const val ID = 5
    }
}
